cpp/src/reductions/all.cu

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// The translation unit for reduction `max`
+
+#include "reduction_functions.hpp"
+#include "simple.cuh"
+
+std::unique_ptr<cudf::scalar> cudf::experimental::reduction::all( column_view const& col, cudf::data_type const output_dtype, cudaStream_t stream,
```

Comment:
Putting every top-level simple reduction function in its own .cu file means you are spending more lines of code on the license header than on the code...  Can you consolidate them all into a single `reduction.cu`?

Reply:
since operator is a template parameter, each operation takes time to compile. so koba seperated each operator to its own .cu file to parallize the compile time. Consolidated reduction.cu will take huge time to compile.

Reply:
Yes, now I recall. Thanks.